.github/workflows/ci.yml

```
@@ -0,0 +1,73 @@
+name: CI
+on: [push, pull_request]
+jobs:
+  test:
```

Internal PRs (that come from branches of pypa/warehouse) match both the push and pull_request events. This leads to a duplication of ci workflows checks on such PRs. Not only is it a waste of resources, it makes the Status Checks UI more frustrating to go through.
```suggestion
  test:
    # We want to run on external PRs, but not on our own internal PRs (for pull_request events) as 
    # they'll be run by the push to the branch. Without this if check, checks are duplicated since 
    # internal PRs match both the push and pull_request events.
    if: github.event_name == 'push' || github.event.pull_request.head.repo.full_name != github.repository
```

---

@di this thing will avoid unnecessary runs for dependabot PRs because it makes pushes to upstream.

---

I think we want to retain the existing behavior.

---

For any pushes? I often add exclude-branches matching dependabot because it always causes PR builds.

---

Yep, for all pushes.